compiler/rustc_middle/src/ty/list.rs

```diff
@@ -43,6 +44,10 @@ extern "C" {
     type OpaqueListContents;
 }
 
+// `OpaqueListContents` is thread-safe
```

Comment:
`OpaqueListContents` is not thread-safe, it does not actually exist, and is just a trick for the compiler. Should the impl be on `List` rather, and bound on `T: Send/Sync`?

Reply:
I impl it on `List` now, and then we should refer to the auto impl of `Send/Sync`